kbase/service_wizard2
# Devops 1369 - List Status

```
/* 
    module_name     - name of the service module
    version         - semantic version number of the service module
    git_commit_hash - git commit hash of the service module
    release_tags    - list of release tags currently for this service module (dev/beta/release)

    url             - the url of the service

    up              - 1 if the service is up, 0 otherwise
    status          - status of the service as reported by rancher
    health          - health of the service as reported by Rancher

    TODO: 
      add something to return: string last_request_timestamp;
*/
typedef structure {
    string module_name;
    string version;
    string git_commit_hash;

    list <string> release_tags;

    string hash;

    string url;

    boolean up;
    string status;
    string health;
} ServiceStatus;
```



```
from ServiceWizardClient import ServiceWizard
sw = ServiceWizard('https://ci.kbase.us/services/service_wizard')
params={"is_up": 0, "module_names": ["onerepotest"]}
a = sw.list_service_status(params=params)
[{'status': 'active', 'hash': .... snip}
```
or
```
import requests
import json
import pprint

url = "https://ci.kbase.us/services/service_wizard/rpc"

       
# Define the JSON-RPC request payload
payload = {
    "rpc": "1.1",
    "method": "ServiceWizard.list_service_status",
    "id": 22,
    "params": [{
        "is_up": False,
        "module_names": ["onerepotest"]
    }],
}
```

# Testing instructions

* Verify that current SW doesn’t behave correctly and returns everything
```
curl -d '{"method":"ServiceWizard.list_service_status","id":22,"params":[ {"module":"onerepotest" } ] }' https://ci.kbase.us/services/service_wizard
```

* Ensure that docs endpoint works
* https://staging.kbase.us/services/service_wizard2/docs
* Ensure that RPC call works

```
curl -d '{"method":"ServiceWizard.list_service_status","id":22,"params":[ {"module":"onerepotest" } ] }' https://staging.kbase.us/services/service_wizard2/rpc
```

* Ensure tests are reasonable
* Ensure that endpoint returns same info on service wizard 1 and service wizard2 
* Ensure that the same general logic is being followed



So the goal of this pr is ensure that the call is the same between both environments via REST or JSON RPC

 {'status': 'active', 'hash': 'ab901d53beb1300b4114b247fc470045fa7d2f41',
  'url': 'https://ci.kbase.us:443/dynserv/ab901d53beb1300b4114b247fc470045fa7d2f41.PDBHTMLReport', 'up': 1,
  'git_commit_hash': 'ab901d53beb1300b4114b247fc470045fa7d2f41', 'version': '0.0.1', 'health': 'healthy',
  'release_tags': ['beta', 'dev'], 'module_name': 'PDBHTMLReport'}]
